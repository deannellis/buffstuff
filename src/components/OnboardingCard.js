import React, { Component } from "react";

import ChinupForm from "../forms/OnboardingChinupForm";
import WeightsForm from "../forms/OnboardingWeightsForm";

class OnboardingCard extends Component {
	constructor(props) {
		super(props);
		this.state = {
			weights: null,
			chinups: null
		};
	}

	submitWeights = weights => {
		console.log("fug");
		const masterWeights = {
			...weights,
			chinups: this.state.chinups
		};
		this.props.onSubmit(masterWeights);
	};

	submitChinups = chinups => {
		this.setState({
			chinups
		});
	};

	render() {
		return (
			<>
				{this.state.chinups === null ? (
					<ChinupForm submitChinups={this.submitChinups} />
				) : (
					<WeightsForm submitWeights={this.submitWeights} />
				)}
			</>
		);
	}
}

export default OnboardingCard;
