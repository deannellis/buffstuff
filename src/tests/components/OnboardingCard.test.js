import React from "react";
import { shallow } from "enzyme";

import OnboardingCard from "../../components/OnboardingCard";

test("should render Starting Weight Form without props", () => {
	const wrapper = shallow(<OnboardingCard />);
	expect(wrapper).toMatchSnapshot();
});

