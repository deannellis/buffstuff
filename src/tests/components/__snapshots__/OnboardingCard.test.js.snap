// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Starting Weight Form with required props 1`] = `
<Fragment>
  <ChinupForm
    submitChinups={[Function]}
  />
</Fragment>
`;
